diff -aruN a/python/clang/cindex.py b/python/clang/cindex.py
--- a/python/clang/cindex.py	2024-03-17 21:19:17
+++ b/python/clang/cindex.py	2024-03-17 21:18:28
@@ -1441,6 +1557,9 @@
 
         return cursor
 
+    def __hash__(self):
+        return self.hash
+
     def __eq__(self, other):
         return conf.lib.clang_equalCursors(self, other)
 
@@ -3841,7 +3960,7 @@
 
 
 class Config:
-    library_path = None
+    library_path = os.environ.get("LIBCLANG_LIBRARY_PATH", os.path.join(os.path.dirname(os.path.realpath(__file__)), "native"))
     library_file = None
     compatibility_check = True
     loaded = False
@@ -3916,7 +4035,7 @@
         else:
             file = "libclang.so"
 
-        if Config.library_path:
+        if Config.library_path and os.path.isfile(Config.library_path + "/" + file):
             file = Config.library_path + "/" + file
 
         return file
